app/models/parent_object.rb

```diff
@@ -283,6 +286,25 @@ def iiif_manifest
     @iiif_manifest = @iiif_presentation.manifest if iiif_presentation.valid?
   end
 
+  def create_or_delete_manifest
+    if (visibility.eql? "Private") && delete_manifest_log?
+      delete_manifest_log
+    elsif create_manifest_log?
+      create_manifest_log
+    end
+  end
+
+  def delete_manifest_log
+    manifest_logs << { status: "Delete", timestamp: Time.zone.now }
+    self.manifest_checksum = nil
+  end
+
+  def create_manifest_log
+    self.manifest_logs ||= []
+    self.manifest_logs << { status: "Create", timestamp: Time.zone.now }
+    self.manifest_checksum = Digest::SHA1.hexdigest iiif_manifest.to_yaml
```

Comment:
This and generate_manifest_job.rb:48 could be moved to a new `IiifPresentation.checksum` method.

Should it checksum `iiif_manifest.to_json` instead of `to_yaml`, so it matches the checksum of the file on S3? (`to_json(pretty: true)` so they match up with `IiifPresentation.save`.)